### PR TITLE
Name consolidation

### PR DESCRIPTION
**Pull request by Weibye**
These are my attempts at solving the naming-scheme question as answered here: https://github.com/bevyengine/rfcs/pull/12#discussion_r680267479

### Solution
* Rename shapes to adhere to `x3d` and `x2d`  naming scheme where possible
* Shapes that only exists in 2d or 3d does not have a `2d` or `3d` suffix
* Renamed `Box2d` to `Rectangle`

[Changes best seen in this table](https://github.com/aevyrie/rfcs/compare/patch-1...Weibye:name-consolidation?expand=1#diff-ec750729cebb6aedfc315644b1fe7e02c7e2839daf399ec925a274f6f944d87fR77)

### Unresolved questions

* Is the `Capsule2d` required? Would it not be better to use the regular `Capsule` shape but combine it with a `Vec2` or `Point2` for positioning?

I'm not very well experienced with either Bevy nor rust, so let me know if these changes doesn't make much sense at all.

### CODE REVIEW

File: rfcs/12-primitive-shapes.md [hunk 1]
```diff
@@ -72,7 +72,32 @@ Colliders can provide intersection checks against other colliders, as well as ch
 
 ### 3D and 2D
 
-This RFC provides independent 2d and 3d primitives. Recall that the purpose of this is to provide lightweight types, so there are what appear to be duplicates in 2d and 3d, such as `Line` and `Line2d`. Note that the 2d version of a line is smaller than its 3d counterpart because it is only defined in 2d. 3d geometry (or 2d with depth) is assumed to be the default for most cases. The names of the types were chosen with this in mind.
+This RFC provides independent 2d and 3d primitives. Recall that the purpose of this is to provide lightweight types, so there are what appear to be duplicates in 2d and 3d, such as `Line2d` and `Line3d`. Note that the 2d version of a line is smaller than its 3d counterpart because it is only defined in 2d.
+
+| Shape           | 2D              | 3D            |
```
Comment (Weibye) [inline rfcs/12-primitive-shapes.md]: I think this table is useful for getting a complete overview of what is suggested in this RFC, at least to look at the naming scheme as a whole. 

I'm wondering if it would make sense to add a `description` column where we note things like _"A line segment bounded by two points"_ for each of the shapes

File: rfcs/12-primitive-shapes.md [hunk 2]
```diff
@@ -115,54 +140,54 @@ trait Collider2d {
 ### 3D Geometry Types
 
 ```rust
-struct Point(Vec3)
+struct Point3d(Vec3)
```
Comment (Weibye) [inline rfcs/12-primitive-shapes.md:276]: Removal of point will be in a separate PR (ref: https://github.com/bevyengine/rfcs/pull/12#discussion_r680269220)

File: rfcs/12-primitive-shapes.md (hunk 1 above)
Comment (alice-i-cecile) [reply rfcs/12-primitive-shapes.md]: I would like a description column, allowing us to steal this directly for dedicated docs.

File: rfcs/12-primitive-shapes.md [hunk 3]
```diff
@@ -72,7 +72,32 @@ Colliders can provide intersection checks against other colliders, as well as ch
...
+| Quad            | Quad2d          | Quad3d        |
+| Sphere          | -               | Sphere        |
+| Box             | -               | Box           |
+| Cylinder        | -               | Cylinder      |
+| Capsule         | -               | Capsule       |
+| Cone            | -               | Cone          |
+| Wedge           | -               | Wedge         |
+| Torus           | -               | Torus         |
+| Frustrum        | -               | Frustrum      |
+| RegularPolygon  | RegularPolygon  | -             |
```
Comment (alice-i-cecile) [inline rfcs/12-primitive-shapes.md]: The 2D only shapes should come first, since the 2D column is first.

File: rfcs/12-primitive-shapes.md (hunk 1 above)
Comment (alice-i-cecile) [reply rfcs/12-primitive-shapes.md:77]: I would also be very interested in a column for "parameters", but that's probably out of scope.

**Review by alice-i-cecile — COMMENTED** [review]
I have a few nits on presentation, but overall this is dramatically clearer. The table is particularly valuable.

**Comment by Weibye** [comment]
* Addressed presentation feedback
* added description column
* Moved mentions of 2d before 3d to keep it consistent throughout the entire RFC

File: rfcs/12-primitive-shapes.md [hunk 4]
```diff
@@ -70,9 +70,38 @@ For use in spatial queries, broad phase collision detection, and raycasting, hyp
...
+| Quad            | Quad2d          | Quad3d        | A polygon with 4 vertices                                                                |
+| Sphere          | -               | Sphere        | A sphere defined by its radius                                                           |
+| Box             | -               | Box           | A cuboid with six quadrilateral faces, defined by height, width, depth                   |
+| Cylinder        | -               | Cylinder      | A cylinder with its origin at the center of the volume                                   |
+| Capsule         | -               | Capsule       | A capsule with its origin at the center of the volume                                    |
+| Cone            | -               | Cone          | A cone with the origin located at the center of the circular base                        |
+| Wedge           | -               | Wedge         | A ramp with the origin centered on the width, and coincident with the rear vertical wall |
+| Torus           | -               | Torus         | A torous                                                                                 |
```
Comment (alice-i-cecile) [inline rfcs/12-primitive-shapes.md]:
```suggestion
| Torus           | -               | Torus         | A torus, shaped like a donut                                                                           |
```

**Review by alice-i-cecile — APPROVED** [review]

**Review by aevyrie — APPROVED** [review]
Looks good, thank you. Can you address the merge conflicts? I think many implementation details were removed from the rfc.

**Comment by Weibye** [comment]
> Looks good, thank you. Can you address the merge conflicts? I think many implementation details were removed from the rfc.

I'll fix this first thing in the morning :)

**Comment by Weibye** [comment]
Sorry for the delay! Have resolved the merge conflict now and this should be good to go :)
